### PR TITLE
Add required form information in `.visually-hidden` span

### LINKED ISSUE #2113: * for required fields is a CSS generated content
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

Star "\*" inside a form label to mention a mandatory field is generated by a CSS class via pseudo element ":after" and property "content:". 
CSS generated content is not well supported by AT/browser see: https://a11ysupport.io/tech/css/generated_content
So we have to change this with a real "*" in a <span> with the specific CSS class "is-réequired" modified, e.g., 
 <div class="col-md-4">
    <label for="validationCustom02" class="form-label">Last name <span class="is-required">*</span></label>
    <input type="text" class="form-control" id="validationCustom02" value="Otto" required>

### Reduced test cases

https://boosted.orange.com/docs/5.2/forms/validation/

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome, Firefox, Microsoft Edge

### What version of Boosted are you using?

5.2

### PR DESCRIPTION
**Pull request by MewenLeHo**
### Related issues

Closes #2113 

### Description

Back to v4.6: https://boosted.orange.com/docs/4.6/components/forms/#custom-styles

I added  a `<span class="visually-hidden"> (required)</span>` information inside each label to provide the `required` information without the need to add an hardcoded `*`

### Motivation & Context

Improve a11y in forms.

### Types of change

- Bug fix (non-breaking which fixes an issue)

### Live previews

- <https://deploy-preview-2114--boosted.netlify.app/docs/5.3/examples/form/>
- <https://deploy-preview-2114--boosted.netlify.app/docs/5.3/forms/overview/#required-field>
- <https://deploy-preview-2114--boosted.netlify.app/docs/5.3/forms/validation/#custom-styles>

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

#### Accessibility

- [x] My change follows accessibility good practices; I have at least run axe

#### Design

- (na) My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
- (na) My change is compatible with responsive display

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
- (na) I have added JavaScript unit tests to cover my changes
- (na) I have added SCSS unit tests to cover my changes

#### Documentation

- [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- [x] My change introduces changes to the migration guide
- (na) My new component is added in Storybook
- (na) My new component is compatible with RTL
- [x] Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
- [x] Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
- [x] Code review
- [x] Design review
- [x] A11y review

#### After the merge

- [ ] Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

File: site/content/docs/5.3/examples/form/index.html
Comment (julien-deramond) [inline site/content/docs/5.3/examples/form/index.html:96]: FYI `.sr-only` was used in Boosted v4, now it's `.visually-hidden`.

**Comment by Aniort** [comment]
good solution, it's OK

**Review by julien-deramond — CHANGES_REQUESTED** [review]
What about the other pages in Boosted documentation like https://boosted.orange.com/docs/5.3/forms/overview/?#required-field?

File: site/content/docs/5.3/examples/form/index.html
Comment (MewenLeHo) [reply site/content/docs/5.3/examples/form/index.html:96]: Yes. And I did it on purpose.

Just like in v4.6: https://boosted.orange.com/docs/4.6/components/forms/#custom-styles
`<label for="validationCustom01" class="is-required">First name<span class="sr-only"> (required)</span></label>`

Without the whitespace, screen readers may have some difficulties to render `Titlerequired` especially in some languages.

I am just wondering if the whitespace should be outside the span:
`<label for="selectTitle" id="selectTitleLabel" class="form-label is-required">Title <span class="visually-hidden">required</span></label>`

Or inside:
`<label for="selectTitle" id="selectTitleLabel" class="form-label is-required">Title<span class="visually-hidden"> required</span></label>`

And I am not even sure if it trully matter 🤔

Comment (julien-deramond) [reply site/content/docs/5.3/examples/form/index.html:96]: In v4.6 the whitespace was included in the visually hidden part so had no effect on the visual rendering which is not the case with your current solution which breaks the rendering by displaying 1 extra whitespace. So inside the `<span>`, if it works for the screen readers, is a better option in terms of visual rendering.

**Comment by MewenLeHo** [comment]
On hold until we receive informations about the A11y team about the best way to write `required` inside labels.

File: site/content/docs/5.3/examples/form/index.html
Comment (louismaximepiton) [inline site/content/docs/5.3/examples/form/index.html:94]: What would you think about make this div `aria-hidden="true"` since it doesn't bring much info to AT users.

Comment (MewenLeHo) [reply site/content/docs/5.3/examples/form/index.html:94]: I am not in favor of it.
The goal here is to give the information to users before they encounter this particular symbol so it's a part of a global 'error prevention strategy'

Comment (louismaximepiton) [reply site/content/docs/5.3/examples/form/index.html:94]: I'm fine with it, my only concern here is: if I'm a user running a not so recent AT that doesn't read CSS content, the AT announce `star indates required` and then I only hear `required` everywhere, so why would I need the first information ? On the other side, the recent ATs read CSS and so `label required star` and if we avoid the first information it could seem weird. I don't know if this is understandable,don't hesitate to ping me if not.

File: site/content/docs/5.3/forms/validation.md
Comment (louismaximepiton) [inline site/content/docs/5.3/forms/validation.md:286]: The label doesn't have the `*`

**Review by louismaximepiton — APPROVED** [review]
👌

**Comment by julien-deramond** [comment]
Please update the description:
* Remove the following part that has been taken into account: "Please note that I only ma[...]"
* Check all the checkboxes and/or use strikethrough

Don't we need a migration guide note in this PR?

**Review by julien-deramond — CHANGES_REQUESTED** [review]
Modifications LGTM! Few things before merging it.

Please update the description:
* Remove the following part that has been taken into account: "Please note that I only ma[...]"
* Check all the checkboxes and/or use strikethrough
* Add the missing live previews

Don't we need a migration guide note in this PR?

Possible enhancement, in https://deploy-preview-2114--boosted.netlify.app/docs/5.3/forms/overview/#required-field, add an a11y callout to talk about why `<span class="visually-hidden"> (required)</span>` is needed and why we made this choice.

**Review by julien-deramond — APPROVED** [review]

**Pull request merged**
